web3/_utils/method_formatters.py

```diff
@@ -486,6 +487,58 @@ def apply_list_to_array_formatter(formatter: Any) -> Callable[..., Any]:
 ABI_REQUEST_FORMATTERS = abi_request_formatters(STANDARD_NORMALIZERS, RPC_ABIS)
 
 
+def get_revert_reason(response: RPCResponse) -> str:
+    """
+    Parse revert reason from response, return None if no revert happened.
+
+    If a revert happened, but no message has been given, return an empty string.
+
+    Reverts contain a `data` attribute with the following layout:
+        "Reverted "
+        Function selector for Error(string): 08c379a (4 bytes)
+        Data offset: 32 (32 bytes)
+        String length (32 bytes)
+        Reason strong (padded, use string length from above to get meaningful part)
+
+    See also https://solidity.readthedocs.io/en/v0.6.3/control-structures.html#revert
+    """
+    if not isinstance(response['error'], dict):
+        return None
+
+    # Parity/OpenEthereum case:
+    data = response['error'].get('data', '')
+    if data.startswith('Reverted '):
+        # "Reverted", function selector and offset are always the same for revert errors
+        prefix = 'Reverted 0x08c379a00000000000000000000000000000000000000000000000000000000000000020'  # noqa: 501
+        if not data.startswith(prefix):
+            return ''
+
+        reason_length = int(data[len(prefix):len(prefix) + 64], 16)
+        reason = data[len(prefix) + 64:len(prefix) + 64 + reason_length * 2]
+        return f'execution reverted: {bytes.fromhex(reason).decode("utf8")}'
+
+    # Geth case:
+    if 'message' in response['error'] and response['error'].get('code', '') == 3:
+        return response['error']['message']
+
+    return None
```

Comment:
gentle nudge to [avoid patterns that require return value checking](https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/antipatterns.md#return-value-checking)

Reply:
Also, `mypy` should be yelling at you about this since this function is designated to return a `str` so it is probably worth looking into why those checks aren't being enforced.